pkg/sessionctx/variable/sysvar_test.go

```diff
@@ -1631,3 +1631,113 @@ func TestTiDBLowResTSOUpdateInterval(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, "1000", val)
 }
+
+func TestTiDBSchemaCacheSize(t *testing.T) {
+	vars := NewSessionVars(nil)
+	mock := NewMockGlobalAccessor4Tests()
+	mock.SessionVars = vars
+	vars.GlobalVarsAccessor = mock
+	var (
+		mb       uint64 = 1 << 20
+		err      error
+		val      string
+		maxValue uint64 = math.MaxInt64
+	)
+	// Test tidb_schema_cache_size
+	schemaCacheSize := GetSysVar(TiDBSchemaCacheSize)
+	// Check default value
+	require.Equal(t, schemaCacheSize.Value, strconv.Itoa(DefTiDBSchemaCacheSize))
+
+	// MinValue is 512 MB
+	err = mock.SetGlobalSysVar(context.Background(), TiDBSchemaCacheSize, strconv.FormatUint(100*mb, 10))
+	require.NoError(t, err)
+	val, err = mock.GetGlobalSysVar(TiDBSchemaCacheSize)
+	require.NoError(t, err)
+	require.Equal(t, "512MB", val)
+
+	// MaxValue is 9223372036854775807
+	err = mock.SetGlobalSysVar(context.Background(), TiDBSchemaCacheSize, strconv.FormatUint(maxValue, 10))
+	require.NoError(t, err)
+	val, err = mock.GetGlobalSysVar(TiDBSchemaCacheSize)
+	require.NoError(t, err)
+	require.Equal(t, strconv.FormatUint(maxValue, 10), val)
+
+	// test MinValue-1
+	err = mock.SetGlobalSysVar(context.Background(), TiDBSchemaCacheSize, strconv.FormatUint(100*mb-1, 10))
+	require.NoError(t, err)
+	val, err = mock.GetGlobalSysVar(TiDBSchemaCacheSize)
+	require.NoError(t, err)
+	require.Equal(t, "512MB", val)
+
+	// test MaxValue+1
+	err = mock.SetGlobalSysVar(context.Background(), TiDBSchemaCacheSize, strconv.FormatUint(maxValue+1, 10))
+	require.NoError(t, err)
+	val, err = mock.GetGlobalSysVar(TiDBSchemaCacheSize)
+	require.NoError(t, err)
+	require.Equal(t, strconv.FormatUint(maxValue, 10), val)
```

Comment:
strange I can't pass this test case in macOS M1

```
    sysvar_test.go:1677: 
        	Error Trace:	/Users/lance6716/Projects/tidb/pkg/sessionctx/variable/sysvar_test.go:1677
        	Error:      	Not equal: 
        	            	expected: "9223372036854775807"
        	            	actual  : "9223372036854775808"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-9223372036854775807
        	            	+9223372036854775808
        	Test:       	TestTiDBSchemaCacheSize
--- FAIL: TestTiDBSchemaCacheSize (0.00s)
```

seems the overflow behaviour is platform specific

https://go.dev/ref/spec#Conversions then search for "Conversions between numeric types"

> In all non-constant conversions involving floating-point or complex values, if the result type cannot represent the value the conversion succeeds but the result value is implementation-dependent.

I think it's safer to check overflow first, but I don't find a easy way to implement it using `units.RAMInBytes` 😂 Sorry for my suggestion. Now I recommend to revert to our `parseByteSize`

Reply:
Anaway, thank you for your review. I have also learned that handling floats with `RAMInBytes` carries some risk of overflow. I reverted to our `parseByteSize`.